mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py

```diff
@@ -162,127 +57,129 @@ class mgmtsystem_nonconformity(base_state, orm.Model):
+    )
+    origin_ids = fields.Many2many(
+        'mgmtsystem.nonconformity.origin',
+        'mgmtsystem_nonconformity_origin_rel',
+        'nonconformity_id',
+        'origin_id',
+        'Origin',
+        required=True,
+    )
+    procedure_ids = fields.Many2many(
+        'document.page',
+        'mgmtsystem_nonconformity_procedure_rel',
+        'nonconformity_id',
+        'procedure_id',
+        'Procedure',
+    )
+    description = fields.Text('Description', required=True)
+    state = fields.Selection(_STATES, 'State', readonly=True, default="draft")
+    state_name = fields.Char(
+        compute='_state_name',
+        string='State Description',
+        size=40,
+    )
+    system_id = fields.Many2one('mgmtsystem.system', 'System')
+
+    # 2. Root Cause Analysis
+    cause_ids = fields.Many2many(
+        'mgmtsystem.nonconformity.cause',
+        'mgmtsystem_nonconformity_cause_rel',
+        'nonconformity_id',
+        'cause_id',
+        'Cause',
+    )
+    severity_id = fields.Many2one(
+        'mgmtsystem.nonconformity.severity',
+        'Severity',
+    )
+    analysis = fields.Text('Analysis')
+    immediate_action_id = fields.Many2one(
+        'mgmtsystem.action',
+        'Immediate action',
+        domain="[('nonconformity_id', '=', id)]",
+    )
+    analysis_date = fields.Datetime('Analysis Date', readonly=True)
+    analysis_user_id = fields.Many2one(
+        'res.users',
+        'Analysis by',
+        readonly=True,
+    )
+
+    # 3. Action Plan
+    action_ids = fields.Many2many(
+        'mgmtsystem.action',
+        'mgmtsystem_nonconformity_action_rel',
+        'nonconformity_id',
+        'action_id',
+        'Actions',
+    )
+    actions_date = fields.Datetime('Action Plan Date', readonly=True)
+    actions_user_id = fields.Many2one(
+        'res.users',
+        'Action Plan by',
+        readonly=True,
+    )
+    action_comments = fields.Text(
+        'Action Plan Comments',
+        help="Comments on the action plan.",
+    )
+
+    # 4. Effectiveness Evaluation
+    evaluation_date = fields.Datetime('Evaluation Date', readonly=True)
+    evaluation_user_id = fields.Many2one(
+        'res.users',
+        'Evaluation by',
+        readonly=True,
+    )
+    evaluation_comments = fields.Text(
+        'Evaluation Comments',
+        help="Conclusions from the last effectiveness evaluation.",
+    )
+
+    # Multi-company
+    company_id = fields.Many2one('res.company', 'Company', default=own_company)
+
+    @api.model
+    def create(self, vals):
         vals.update({
-            'ref': self.pool.get('ir.sequence').get(
-                cr, uid, 'mgmtsystem.nonconformity')
+            'ref': self.env['ir.sequence'].get('mgmtsystem.nonconformity')
         })
-        return super(mgmtsystem_nonconformity, self).create(
-            cr, uid, vals, context)
+        return super(mgmtsystem_nonconformity, self).create(vals)
 
     def message_auto_subscribe(
```

Comment:
The messaging features need a rewrite. I'm not bothered by the direct port, but wouldn't mind if they were removed now.